vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ML_MODEL_LINEAR_LINEAR_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ML_MODEL_LINEAR_LINEAR_H_
+
+#include <map>
+#include <string>
+
+#include "bat/ads/internal/ml/data/vector_data.h"
+#include "bat/ads/internal/ml/ml_aliases.h"
+
+namespace ads {
+namespace ml {
+namespace model {
+
+class Linear {
+ public:
+  Linear();
+
+  Linear(const Linear& other);
+
+  explicit Linear(const std::string& model);
+
+  Linear(const std::map<std::string, data::VectorData>& weights,
+         const std::map<std::string, double>& biases);
+
+  ~Linear();
+
+  PredictionMap Predict(const data::VectorData& x);
+
+  PredictionMap TopPredictions(const data::VectorData& x, int top_count = -1);

```

Comment:
make `const`, nit: maybe rename to `GetTopPredictions`

Reply:
Done